rfcs/52-unstable_features.md

```diff
@@ -0,0 +1,65 @@
+- Something in between that would benefit from iterative work? Here come the unstable feature workflow for you!
+
+You will need to open a tracking issue to explain the new feature you want to add. If it looks good, you'll be ready to open that PR with a feature gate for your changes!
+
+## Implementation strategy
+
+The new workflow for this kind of feature would be:
+- Submit an issue explaining the feature you want to add under a feature gate. This will be the tracking issue for stabilization once approved. This issue gets a new label `S-Stabilization`.
+- If approved, submit a PR with the new feature and the feature gate
+- This PR can be merged by someone with the merge rights on the related part of Bevy as soon as it's `S-Ready-For-Final-Review`, without an approval by the main Bevy maintainers.
+- Subsequent PRs on the feature must be linked to the tracking issue and follow the same process.
+- Once the feature is finished, enter the stabilization period. Open a PR that will remove the feature gate. This PR need to be approved by the main Bevy maintainers.
+
+An existing PR can be retroffited in this process on the suggestion of Bevy community members, and if the PR author agrees.
+
+The RFCs and the unstable features are closely related. An RFC can be implemented as an unstable feature to help resolve issues that need more experimentation. An unstable feature could require an RFC to reach stabilization.
+
+Each PR will still need to be approved by Bevy community members, and would still need to have the same level of quality as other PRs merged.
+
+The PRs for the initial feature, various changes during its finalization and for the stabilization can be opened by different persons.
+
+The feature gate should not be exposed on the main `bevy` crate, but only on the subcrates where it is relevant. It can be enabled as a user by adding dependencies directly on the subcrates and enabling the feature.
+
+## Drawbacks
+
+This will increase complexity in Bevy code.
+
+It will also increase complexity of a contribution that would like to use this process, but it is opt in and not mandatory. Think of it as a fast track for unstable features.
+
+## Rationale and alternatives
+
+- Improve the process to get new features in Bevy that would benefir from iterative work.
+- Alternatives:
+    - Continue as it is now, with semi large PRs that bitrot and RFCs that don't often move forward
+    - Work harder to implement a new feature as a third party crate (this is not always possible)
+    - Fork Bevy to merge unnaproved PRs and check how they work / interact
+
+
+## Prior art
+
+[The Rust process](https://rustc-dev-guide.rust-lang.org/implementing_new_features.html)
+
+## Unresolved questions
+
+- Do we want to limit the number of feature gates in Bevy to reduce complexity?
+    - this could help control the increased complexity in Bevy
+- Do we want to limit the maximum age of a unstabilized feature?
```

Comment:
With the Bevy train release schedule, it'd be nice to have a minimum time for this too, to allow for folks using the released versions of Bevy to have to test out and file bugs. This could be mitigated by telling users to use the `main` branch, but that might not always be possible with the growing ecosystem and dependency compatibility.

As an upper bound, if a unstable feature doesn't get stabilized within two train releases, even with external users testing it out, I'd say we either need to reconsider the design of it as a full RFC or just scrap the idea entirely.

Reply:
With Rust as an example where stabilising a feature can take a few years, I think a good scope would be at least one version, at most 4 (so between 3 months and a year)

Reply:
I like both the upper and lower bounds here. The upper bound in particular results in a nice forcing function to actually ship things or cut our losses.